TestDxfDocument.test_spline fails with unsupported operand

Dear all,

I try to compile cadquery (Commit: d780127) on arm64. I already managed to get to compiling cadquery but one of the tests fails:

```
 ============================= test session starts ==============================
platform linux -- Python 3.10.12, pytest-7.4.2, pluggy-1.3.0 -- $PREFIX/bin/python3.10

[...]

=================================== FAILURES ===================================
_________________________ TestDxfDocument.test_spline __________________________

self = <tests.test_exporters.TestDxfDocument testMethod=test_spline>

     def test_spline(self):
         pts = [(0, 0), (0, 0.5), (1, 1)]
         workplane = (
             Workplane().spline(pts).close().extrude(1).edges("|Z").fillet(0.1).section()
         )

         plane = workplane.plane
         shape = toCompound(workplane).transformShape(plane.fG)
         edges = shape.Edges()

         result_type, result_attributes = DxfDocument._dxf_spline(edges[0], plane)

         expected_type, expected_attributes = (
             "SPLINE",
             {
                 "control_points": [
                     (-0.032010295564216654, 0.2020130195642037, 0.0),
                     (-0.078234124721739, 0.8475143728081896, 0.0),
                     (0.7171193004814275, 0.9728923786984539, 0.0),
                 ],
                 "order": 3,
                 "knots": [
                     0.18222956891558767,
                     0.18222956891558767,
                     0.18222956891558767,
                     1.416096480384525,
                     1.416096480384525,
                     1.416096480384525,
                 ],
                 "weights": None,
             },
         )

         self.assertEqual(expected_type, result_type)
 >       self.assertAlmostEqual(
             expected_attributes["control_points"], result_attributes["control_points"]
         )
E       TypeError: unsupported operand type(s) for -: 'list' and 'list'

 tests/test_exporters.py:418: TypeError
```

I assume that this is not an arm-specific issue. I found some topics on the web, that almost equal doesn't support lists of floats. I wonder why this test seems to be passing in the official build pipeline. Any idea what could be the reason in my case?

Best regards

Tank you for your investigation. I ran the build process with suggested modification. It seems to be working, but now another failure arises. I think this might be somehow related as basically it seems to be a floating point precision issue (as also for not triggering the mentioned shortcut):

```
#31 198.9 =================================== FAILURES ===================================
#31 198.9 _________________________ TestDxfDocument.test_spline __________________________
#31 198.9 
#31 198.9 self = <tests.test_exporters.TestDxfDocument testMethod=test_spline>
#31 198.9 
#31 198.9     def test_spline(self):
#31 198.9         pts = [(0, 0), (0, 0.5), (1, 1)]
#31 198.9         workplane = (
#31 198.9             Workplane().spline(pts).close().extrude(1).edges("|Z").fillet(0.1).section()
#31 198.9         )
#31 198.9     
#31 198.9         plane = workplane.plane
#31 198.9         shape = toCompound(workplane).transformShape(plane.fG)
#31 198.9         edges = shape.Edges()
#31 198.9     
#31 198.9         result_type, result_attributes = DxfDocument._dxf_spline(edges[0], plane)
#31 198.9     
#31 198.9         expected_type, expected_attributes = (
#31 198.9             "SPLINE",
#31 198.9             {
#31 198.9                 "control_points": [
#31 198.9                     (-0.032010295564216654, 0.2020130195642037, 0.0),
#31 198.9                     (-0.078234124721739, 0.8475143728081896, 0.0),
#31 198.9                     (0.7171193004814275, 0.9728923786984539, 0.0),
#31 198.9                 ],
#31 198.9                 "order": 3,
#31 198.9                 "knots": [
#31 198.9                     0.18222956891558767,
#31 198.9                     0.18222956891558767,
#31 198.9                     0.18222956891558767,
#31 198.9                     1.416096480384525,
#31 198.9                     1.416096480384525,
#31 198.9                     1.416096480384525,
#31 198.9                 ],
#31 198.9                 "weights": None,
#31 198.9             },
#31 198.9         )
#31 198.9     
#31 198.9         self.assertEqual(expected_type, result_type)
#31 198.9         for expected, result in zip(expected_attributes["control_points"], result_attributes["control_points"]):
#31 198.9             assert expected == approx(result)
#31 198.9         self.assertEqual(expected_attributes["order"], result_attributes["order"])
#31 198.9 >       self.assertEqual(expected_attributes["knots"], result_attributes["knots"])
#31 198.9 E       AssertionError: Lists differ: [0.18[55 chars]67, 1.416096480384525, 1.416096480384525, 1.416096480384525] != [0.18[55 chars]67, 1.4160964803845248, 1.4160964803845248, 1.4160964803845248]
#31 198.9 E       
#31 198.9 E       First differing element 3:
#31 198.9 E       1.416096480384525
#31 198.9 E       1.4160964803845248
#31 198.9 E       
#31 198.9 E         [0.18222956891558767,
#31 198.9 E          0.18222956891558767,
#31 198.9 E          0.18222956891558767,
#31 198.9 E       -  1.416096480384525,
#31 198.9 E       ?                  ^
#31 198.9 E       
#31 198.9 E       +  1.4160964803845248,
#31 198.9 E       ?                  ^^
#31 198.9 E       
#31 198.9 E       -  1.416096480384525,
#31 198.9 E       ?                  ^
#31 198.9 E       
#31 198.9 E       +  1.4160964803845248,
#31 198.9 E       ?                  ^^
#31 198.9 E       
#31 198.9 E       -  1.416096480384525]
#31 198.9 E       ?                  ^
#31 198.9 E       
#31 198.9 E       +  1.4160964803845248]
#31 198.9 E       ?                  ^^
#31 198.9 
#31 198.9 tests/test_exporters.py:421: AssertionError
```

--- a/tests/test_exporters.py
+++ b/tests/test_exporters.py
@@ -415,11 +415,14 @@ class TestDxfDocument(BaseTest):
         )
 
         self.assertEqual(expected_type, result_type)
-        self.assertAlmostEqual(
+
+        for expected, result in zip(
             expected_attributes["control_points"], result_attributes["control_points"]
-        )
+        ):
+            assert result == approx(expected)
+
         self.assertEqual(expected_attributes["order"], result_attributes["order"])
-        self.assertEqual(expected_attributes["knots"], result_attributes["knots"])
+        assert result_attributes["knots"] == approx(expected_attributes["knots"])
         self.assertEqual(expected_attributes["weights"], result_attributes["weights"])
 
     def test_add_layer_definition(self):

